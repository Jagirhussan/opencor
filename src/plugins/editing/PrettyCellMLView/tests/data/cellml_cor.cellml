<?xml version="1.0" encoding="utf-8"?>
<model name="cellml_1_0" xmlns="http://www.cellml.org/cellml/1.0#" xmlns:cellml="http://www.cellml.org/cellml/1.0#" xmlns:cmeta="http://www.cellml.org/metadata/1.0#">
    <units base_units="yes" name="baseunit1"/>
    <units base_units="yes" name="baseunit2"/>
    <units base_units="yes" name="baseunit3"/>
    <units name="unit1">
        <unit units="ampere"/>
        <unit units="becquerel"/>
        <unit units="candela"/>
        <unit units="celsius"/>
        <unit units="coulomb"/>
        <unit units="dimensionless"/>
        <unit units="farad"/>
        <unit units="gram"/>
        <unit units="gray"/>
        <unit units="henry"/>
        <unit units="hertz"/>
        <unit units="joule"/>
        <unit units="katal"/>
        <unit units="kelvin"/>
        <unit units="kilogram"/>
        <unit units="liter"/>
        <unit units="litre"/>
        <unit units="lumen"/>
        <unit units="lux"/>
        <unit units="meter"/>
        <unit units="metre"/>
        <unit units="mole"/>
        <unit units="newton"/>
        <unit units="ohm"/>
        <unit units="pascal"/>
        <unit units="radian"/>
        <unit units="second"/>
        <unit units="siemens"/>
        <unit units="sievert"/>
        <unit units="steradian"/>
        <unit units="tesla"/>
        <unit units="volt"/>
        <unit units="watt"/>
        <unit units="weber"/>
    </units>
    <units name="unit2">
        <unit exponent="-1" multiplier="3" offset="7" prefix="milli" units="someunit"/>
    </units>
    <units name="unit3">
        <unit prefix="yotta" units="someunit"/>
        <unit prefix="zetta" units="someunit"/>
        <unit prefix="exa" units="someunit"/>
        <unit prefix="peta" units="someunit"/>
        <unit prefix="tera" units="someunit"/>
        <unit prefix="giga" units="someunit"/>
        <unit prefix="mega" units="someunit"/>
        <unit prefix="kilo" units="someunit"/>
        <unit prefix="hecto" units="someunit"/>
        <unit prefix="deka" units="someunit"/>
        <unit prefix="deci" units="someunit"/>
        <unit prefix="centi" units="someunit"/>
        <unit prefix="milli" units="someunit"/>
        <unit prefix="micro" units="someunit"/>
        <unit prefix="nano" units="someunit"/>
        <unit prefix="milli" units="someunit"/>
        <unit prefix="pico" units="someunit"/>
        <unit prefix="femto" units="someunit"/>
        <unit prefix="atto" units="someunit"/>
        <unit prefix="zepto" units="someunit"/>
        <unit prefix="yocto" units="someunit"/>
    </units>
    <component name="component1">
        <units base_units="yes" name="baseunit1"/>
        <units base_units="yes" name="baseunit2"/>
        <units base_units="yes" name="baseunit3"/>
        <units name="unit1">
            <unit units="ampere"/>
            <unit units="becquerel"/>
            <unit units="candela"/>
            <unit units="celsius"/>
            <unit units="coulomb"/>
            <unit units="dimensionless"/>
            <unit units="farad"/>
            <unit units="gram"/>
            <unit units="gray"/>
            <unit units="henry"/>
            <unit units="hertz"/>
            <unit units="joule"/>
            <unit units="katal"/>
            <unit units="kelvin"/>
            <unit units="kilogram"/>
            <unit units="liter"/>
            <unit units="litre"/>
            <unit units="lumen"/>
            <unit units="lux"/>
            <unit units="meter"/>
            <unit units="metre"/>
            <unit units="mole"/>
            <unit units="newton"/>
            <unit units="ohm"/>
            <unit units="pascal"/>
            <unit units="radian"/>
            <unit units="second"/>
            <unit units="siemens"/>
            <unit units="sievert"/>
            <unit units="steradian"/>
            <unit units="tesla"/>
            <unit units="volt"/>
            <unit units="watt"/>
            <unit units="weber"/>
        </units>
        <units name="unit2">
            <unit exponent="-1" multiplier="3" offset="7" prefix="milli" units="someunit"/>
        </units>
        <units name="unit3">
            <unit prefix="yotta" units="someunit"/>
            <unit prefix="zetta" units="someunit"/>
            <unit prefix="exa" units="someunit"/>
            <unit prefix="peta" units="someunit"/>
            <unit prefix="tera" units="someunit"/>
            <unit prefix="giga" units="someunit"/>
            <unit prefix="mega" units="someunit"/>
            <unit prefix="kilo" units="someunit"/>
            <unit prefix="hecto" units="someunit"/>
            <unit prefix="deka" units="someunit"/>
            <unit prefix="deci" units="someunit"/>
            <unit prefix="centi" units="someunit"/>
            <unit prefix="milli" units="someunit"/>
            <unit prefix="micro" units="someunit"/>
            <unit prefix="nano" units="someunit"/>
            <unit prefix="milli" units="someunit"/>
            <unit prefix="pico" units="someunit"/>
            <unit prefix="femto" units="someunit"/>
            <unit prefix="atto" units="someunit"/>
            <unit prefix="zepto" units="someunit"/>
            <unit prefix="yocto" units="someunit"/>
        </units>
        <variable name="variable1" units="unit1"/>
        <variable initial_value="123" name="variable2" units="unit2"/>
        <variable name="variable3" public_interface="in" units="unit3"/>
        <variable name="variable4" private_interface="out" units="unit4"/>
        <variable name="variable5" public_interface="none" private_interface="none" units="unit5"/>
        <math xmlns="http://www.w3.org/1998/Math/MathML">
            <apply>
                <eq/>
                <ci>a</ci>
                <ci>b</ci>
            </apply>
            <apply>
                <eq/>
                <ci>c</ci>
                <cn cellml:units="dimensionless">3</cn>
            </apply>
            <apply>
                <eq/>
                <ci>d</ci>
                <cn>5</cn>
            </apply>
            <apply>
                <eq/>
                <ci>e</ci>
                <cn cellml:units="dimensionless" type="e-notation">12.3<sep/>5</cn>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <eq/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <neq/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <gt/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <lt/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <geq/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <leq/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <minus/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <times/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <divide/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <power/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <power/>
                    <ci>b</ci>
                    <cn cellml:units="dimensionless">3</cn>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <power/>
                    <ci>b</ci>
                    <cn cellml:units="dimensionless">2</cn>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <power/>
                    <ci>b</ci>
                    <cn cellml:units="dimensionless">0.5</cn>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <root/>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <root/>
                    <ci>b</ci>
                    <cn cellml:units="dimensionless">2</cn>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <root/>
                    <ci>b</ci>
                    <ci>c</ci>
                </apply>
            </apply>
        </math>
    </component>
    <group>
        <relationship_ref relationship="containment"/>
        <component_ref component="component1">
            <component_ref component="component11">
                <component_ref component="component111"/>
                <component_ref component="component112"/>
            </component_ref>
        </component_ref>
    </group>
    <group>
        <relationship_ref name="containment1" relationship="containment"/>
        <component_ref component="component1">
            <component_ref component="component12">
                <component_ref component="component121"/>
            </component_ref>
            <component_ref component="component13"/>
        </component_ref>
    </group>
    <group>
        <relationship_ref relationship="encapsulation"/>
        <component_ref component="component11">
            <component_ref component="component111"/>
            <component_ref component="component112"/>
        </component_ref>
        <component_ref component="component12">
            <component_ref component="component121"/>
        </component_ref>
    </group>
    <group>
        <relationship_ref relationship="encapsulation"/>
        <relationship_ref name="containment1" relationship="containment"/>
        <component_ref component="component1"/>
    </group>
    <connection>
        <map_components component_1="component1" component_2="component2"/>
        <map_variables variable_1="variable1" variable_2="variable2"/>
    </connection>
</model>
