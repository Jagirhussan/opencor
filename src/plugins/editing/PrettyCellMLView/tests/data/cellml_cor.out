def model cellml_1_0 as
    def unit baseunit1 as base unit;
    def unit baseunit2 as base unit;
    def unit baseunit3 as base unit;

    def unit unit1 as
        unit ampere;
        unit becquerel;
        unit candela;
        unit celsius;
        unit coulomb;
        unit dimensionless;
        unit farad;
        unit gram;
        unit gray;
        unit henry;
        unit hertz;
        unit joule;
        unit katal;
        unit kelvin;
        unit kilogram;
        unit liter;
        unit litre;
        unit lumen;
        unit lux;
        unit meter;
        unit metre;
        unit mole;
        unit newton;
        unit ohm;
        unit pascal;
        unit radian;
        unit second;
        unit siemens;
        unit sievert;
        unit steradian;
        unit tesla;
        unit volt;
        unit watt;
        unit weber;
    enddef;

    def unit unit2 as
        unit someunit {pref: milli, expo: -1, mult: 3, off: 7};
    enddef;

    def unit unit3 as
        unit someunit {pref: yotta};
        unit someunit {pref: zetta};
        unit someunit {pref: exa};
        unit someunit {pref: peta};
        unit someunit {pref: tera};
        unit someunit {pref: giga};
        unit someunit {pref: mega};
        unit someunit {pref: kilo};
        unit someunit {pref: hecto};
        unit someunit {pref: deka};
        unit someunit {pref: deci};
        unit someunit {pref: centi};
        unit someunit {pref: milli};
        unit someunit {pref: micro};
        unit someunit {pref: nano};
        unit someunit {pref: milli};
        unit someunit {pref: pico};
        unit someunit {pref: femto};
        unit someunit {pref: atto};
        unit someunit {pref: zepto};
        unit someunit {pref: yocto};
    enddef;

    def comp component1 as
        def unit baseunit1 as base unit;
        def unit baseunit2 as base unit;
        def unit baseunit3 as base unit;

        def unit unit1 as
            unit ampere;
            unit becquerel;
            unit candela;
            unit celsius;
            unit coulomb;
            unit dimensionless;
            unit farad;
            unit gram;
            unit gray;
            unit henry;
            unit hertz;
            unit joule;
            unit katal;
            unit kelvin;
            unit kilogram;
            unit liter;
            unit litre;
            unit lumen;
            unit lux;
            unit meter;
            unit metre;
            unit mole;
            unit newton;
            unit ohm;
            unit pascal;
            unit radian;
            unit second;
            unit siemens;
            unit sievert;
            unit steradian;
            unit tesla;
            unit volt;
            unit watt;
            unit weber;
        enddef;

        def unit unit2 as
            unit someunit {pref: milli, expo: -1, mult: 3, off: 7};
        enddef;

        def unit unit3 as
            unit someunit {pref: yotta};
            unit someunit {pref: zetta};
            unit someunit {pref: exa};
            unit someunit {pref: peta};
            unit someunit {pref: tera};
            unit someunit {pref: giga};
            unit someunit {pref: mega};
            unit someunit {pref: kilo};
            unit someunit {pref: hecto};
            unit someunit {pref: deka};
            unit someunit {pref: deci};
            unit someunit {pref: centi};
            unit someunit {pref: milli};
            unit someunit {pref: micro};
            unit someunit {pref: nano};
            unit someunit {pref: milli};
            unit someunit {pref: pico};
            unit someunit {pref: femto};
            unit someunit {pref: atto};
            unit someunit {pref: zepto};
            unit someunit {pref: yocto};
        enddef;

        var variable1: unit1;
        var variable2: unit2 {init: 123};
        var variable3: unit3 {pub: in};
        var variable4: unit4 {priv: out};
        var variable5: unit5 {pub: none, priv: none};

        a = b;
        c = 3{dimensionless};
        d = 5{};
        e = 12.3e5{dimensionless};
        a = b == c;
        a = b <> c;
        a = b > c;
        a = b < c;
        a = b >= c;
        a = b <= c;
        a = b;
        a = b+c;
        a = b+c+d+e+f;
        a = -b;
        a = b-c;
        a = b-c-d-e-f;
        a = b*c;
        a = b*c*d*e*f;
        a = b/c;
        a = b/c/d/e/f;
        a = pow(b, c);
        a = sqrt(b);
        a = sqr(b);
        a = pow(b, 3{dimensionless});
        a = sqrt(b);
        a = sqrt(b);
        a = root(b, 3{dimensionless});
        a = root(b, c);
        a = abs(b);
        a = exp(b);
        a = ln(b);
        a = log(b);
        a = log(b, c);
        a = ceil(b);
        a = floor(b);
        a = fact(b);
        a = b and c;
        a = b and c and d and e and f;
        a = b or c;
        a = b or c or d or e or f;
        a = b xor c;
        a = b xor c xor d xor e xor f;
        a = not(b);
        a = not(b or c);
        a = ode(f, t);
        a = ode(f, t, 2{dimensionless});
    enddef;

    def group as containment for
        comp component1 incl
            comp component11 incl
                comp component111;
                comp component112;
            endcomp;
        endcomp;
    enddef;

    def group as containment containment1 for
        comp component1 incl
            comp component12 incl
                comp component121;
            endcomp;

            comp component13;
        endcomp;
    enddef;

    def group as encapsulation for
        comp component11 incl
            comp component111;
            comp component112;
        endcomp;

        comp component12 incl
            comp component121;
        endcomp;
    enddef;

    def group as encapsulation and containment containment1 for
        comp component1;
    enddef;

    def map between component1 and component2 for
        vars variable1 and variable2;
    enddef;
enddef;
